test/extended/olm/olmv1-webhook-support.go

```diff
@@ -0,0 +1,331 @@
+package operators
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/clientcmd"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	conditionTypeServing   = "Serving"
+	conditionTypeInstalled = "Installed"
+
+	openshiftServiceCANamespace            = "openshift-service-ca"
+	openshiftServiceCASigningKeySecretName = "signing-key"
+)
+
+var (
+	webhookTestBaseDir              = exutil.FixturePath("testdata", "olmv1", "webhook-support")
+	webhookOperatorInstallNamespace = "webhook-operator"
+)
+
+var _ = g.Describe("[sig-olmv1][OCPFeatureGate:NewOLMWebhookProviderOpenshiftServiceCA][Skipped:Disconnected][Serial] OLMv1 operator with webhooks", g.Ordered, g.Serial, func() {
+	defer g.GinkgoRecover()
+	oc := exutil.NewCLIWithoutNamespace("openshift-operator-controller-webhooks")
+
+	var cleanup func()
+
+	g.BeforeEach(func(ctx g.SpecContext) {
+		cleanup = setupWebhookOperator(ctx, oc)
+	})
+
+	g.AfterEach(func() {
+		if cleanup != nil {
+			cleanup()
+		}
+		if g.CurrentSpecReport().Failed() {
+			exutil.DumpPodLogsStartingWith("", oc)
+		}
+	})
+
+	g.It("should have a working validating webhook", func(ctx g.SpecContext) {
+		g.By("creating a webhook test resource that will be rejected by the validating webhook")
+		resource := getWebhookOperatorResource("validating-webhook-test", webhookOperatorInstallNamespace, false)
+
+		// even if the resource gets created, it will be cleaned up when test gets torn down and the
+		// webhook-operator namespace gets deleted
+		o.Eventually(func(cg o.Gomega) {
+			_, err := applyRenderedManifests(oc, resource)
+			cg.Expect(err).To(o.HaveOccurred())
+			cg.Expect(err.Error()).To(o.ContainSubstring("Invalid value: false: Spec.Valid must be true"))
+		}).WithTimeout(10 * time.Second).WithPolling(250 * time.Millisecond).Should(o.Succeed())
+	})
+
+	g.It("should have a working mutating webhook", func(ctx g.SpecContext) {
+		g.By("creating a valid webhook test resource")
+		var (
+			mutatingWebhookResourceName = "mutating-webhook-test"
+			resource                    = getWebhookOperatorResource(mutatingWebhookResourceName, webhookOperatorInstallNamespace, true)
+		)
+
+		o.Eventually(func(cg o.Gomega) {
+			cleanup, err := applyRenderedManifests(oc, resource)
+			cg.Expect(err).NotTo(o.HaveOccurred())
+			// clean up fn will only be scheduled if there are no errors
+			g.DeferCleanup(cleanup)
+		}).WithTimeout(10 * time.Second).WithPolling(250 * time.Millisecond).Should(o.Succeed())
```

Comment:
OCP API already suffers from latency, and these tests make many API calls. Why are we doing that? 
I am fraid that causes flakes